ReactNative/ReactTwitter/core/views/tweetsList.js

```diff
@@ -1,20 +1,33 @@
-import React, { Component } from 'react'
-import { ListView } from 'react-native'
+import React from 'react'
+import {
+  ListView,
+  BackAndroid,
+  Navigator
+} from 'react-native'
 import TweetsListItem from './tweetsListItem'
 
-class TweetsList extends Component {
+var TweetsList = React.createClass({
+  propTypes: {
+    navigator: React.PropTypes.instanceOf(Navigator).isRequired
+  },
 
-  constructor (props) {
-    super(props)
+  getInitialState () {
     var ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2})
-    this.state = {
+    return {
       dataSource: ds.cloneWithRows([])
     }
-  }
+  },
 
   componentDidMount () {
+    BackAndroid.addEventListener('hardwareBackPress', () => {
```

Comment:
That's the reason why we should implement this somewhere else, we already have this code duplicated in 2 screens